torch/utils/data/dataloader.py

```diff
@@ -421,12 +443,28 @@ def __init__(
         torch.set_vital("Dataloader", "enabled", "True")  # type: ignore[attr-defined]
 
     def _get_iterator(self) -> _BaseDataLoaderIter:
+        if self.stateful:
+            return self._get_stateful_iterator()
         if self.num_workers == 0:
             return _SingleProcessDataLoaderIter(self)
         else:
             self.check_worker_number_rationality()
             return _MultiProcessingDataLoaderIter(self)
 
+    def _get_stateful_iterator(
+        self,
+    ) -> Union[_StatefulSingleProcessDataLoaderIter,]:
```

Comment:
Union with one type is a bit strange.  Seems like there's an appropriate base class.

```suggestion
    ) -> _StatefulBaseDataLoaderIter:
```